src/MotionSource/Targets/AbstractPersonTarget.cs

```diff
@@ -0,0 +1,279 @@
+﻿using SimpleJSON;
+using System.Collections.Generic;
+using System.Linq;
+using ToySerialController.UI;
+using ToySerialController.Utils;
+using UnityEngine;
+using DebugUtils;
+using System;
+
+namespace ToySerialController.MotionSource
+{
+    public abstract class AbstractPersonTarget : IMotionSourceTarget
+    {
+        protected Atom _personAtom;
+        protected Vector3 _targetPosition;
+        protected Vector3 _targetUp;
+        protected Vector3 _targetRight;
+        protected Vector3 _targetForward;
+
+        private JSONStorableStringChooser PersonChooser;
+        private JSONStorableStringChooser TargetChooser;
+        private JSONStorableFloat TargetOffsetSlider;
+        
+        protected SuperController Controller => SuperController.singleton;
+
+        public IMotionSourceActor Actor { get; set; }
+
+        public Vector3 TargetPosition => _targetPosition + _targetUp * TargetOffsetSlider.val;
+        public Vector3 TargetUp => _targetUp;
+        public Vector3 TargetRight => _targetRight;
+        public Vector3 TargetForward => _targetForward;
+
+        protected IDictionary<string, Func<bool>> TargetUpdaters { get; } = new Dictionary<string, Func<bool>>();
+
+        protected IList<Func<bool>> AutoUpdaters { get; }
```

Comment:
Use `List<>`

Reply:
Why would you prefer a concrete implementation over an agnostic interface?

Reply:
No real need to abstract in this case, its a private list.
Also `IList` will be slightly slower since each method will be virtual.

Reply:
What do you mean? Explicit interface implementations cannot be virtual and for non-explicit implementations it does not make a difference whether you call them by class or by interface.

Reply:
I've never heard of a performance impact from virtual methods, either. It's resolved at compile time and doesn't matter afterwards.
I don't think there's a measurable difference in compile time when vam loads the plugin's code.

Reply:
All interface methods are marked as virtual by default if you look at the IL.
I dont know if there is a language where virtual methods are NOT slower, virtual calls require usage of the vtable, interface methods will probably be even slower due to having to resolve the implementation too. I'm pretty sure vam/unity will not devirtualize anything, maybe in newer .net versions.
Ofc those are marginal differences, but I still dont think there is a point in using interfaces on private fields, maybe on public properties.